// @txtextcontrol/tx-ng-document-editor/esm5/lib/document-editor/color.js
/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes,constantProperty,extraRequire,missingOverride,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
 */
var Color = /** @class */ (function () {
    function Color(r, g, b, a) {
        if (a === void 0) { a = 0xFF; }
        this.R = 0;
        this.G = 0;
        this.B = 0;
        this.A = 0xFF;
        this.R = r;
        this.G = g;
        this.B = b;
        this.A = a;
    }
    /**
     * @return {?}
     */
    Color.prototype.toHexString = /**
     * @return {?}
     */
    function () {
        return "#" + pad(this.R.toString(16)) + pad(this.G.toString(16)) + pad(this.B.toString(16));
    };
    /**
     * @param {?} value
     * @return {?}
     */
    Color.fromHexString = /**
     * @param {?} value
     * @return {?}
     */
    function (value) {
        /** @type {?} */
        var r;
        /** @type {?} */
        var g;
        /** @type {?} */
        var b;
        if (!value || value[0] != '#' || (value.length != 7 && value.length != 4)) {
            throw new Error('String is not a hex color value.');
        }
        if (value.length === 7) {
            r = value.substring(1, 3);
            g = value.substring(3, 5);
            b = value.substring(5, 7);
            return new Color(parseInt(r, 16), parseInt(g, 16), parseInt(b, 16));
        }
        else {
            r = value[1];
            g = value[2];
            b = value[3];
            return new Color(parseInt(r + r, 16), parseInt(g + g, 16), parseInt(b + b, 16));
        }
    };
    return Color;
}());
export { Color };
if (false) {
    /** @type {?} */
    Color.prototype.R;
    /** @type {?} */
    Color.prototype.G;
    /** @type {?} */
    Color.prototype.B;
    /** @type {?} */
    Color.prototype.A;
}
/**
 * @param {?} num
 * @param {?=} size
 * @return {?}
 */
function pad(num, size) {
    if (size === void 0) { size = 2; }
    /** @type {?} */
    var result = num + '';
    while (result.length < size)
        result = '0' + result;
    return result;
}
