// @txtextcontrol/tx-ng-document-editor/esm5/lib/document-editor/tx-settings.js
/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes,constantProperty,extraRequire,missingOverride,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
 */
import { Color } from "./color";
var TXSettings = /** @class */ (function () {
    function TXSettings(WebSocketURL) {
        if (WebSocketURL === void 0) { WebSocketURL = ''; }
        this.WebSocketURL = WebSocketURL;
        this.IsMVC = false;
        this.NoPrintEndpoint = true;
        this.NoFileSystemAccess = true; // No file system access on the web server
        // No file system access on the web server
        this.ConnectionID = '';
        this.DocumentTargetMarkers = true;
        this.BackColor = new Color(0xFF, 0xFF, 0xFF);
        this.StatusBarColor = new Color(0x2B, 0x56, 0x9A);
        this.EditMode = 1;
        this.ContextMenusEnabled = true;
        this.SaveDocumentQuestionDialogEnabled = true;
        this.TableGridLines = true;
        this.TextFrameMarkerLines = true;
        this.ControlChars = false;
        this.TextFieldsEditable = false;
        this.FormattingPrinter = '';
        this.Culture = '';
        this.UICulture = '';
        this.UserNames = [];
        this.ReconnectTimeout = 0;
    }
    /**
     * @param {?} value
     * @return {?}
     */
    TXSettings.editModeFromString = /**
     * @param {?} value
     * @return {?}
     */
    function (value) {
        switch (value) {
            case 'Edit': return 1;
            case 'ReadAndSelect': return 2;
            case 'ReadOnly': return 3;
            case 'UsePassword': return 2048;
        }
        return 1; // EditMode.Edit
    };
    return TXSettings;
}());
export { TXSettings };
if (false) {
    /** @type {?} */
    TXSettings.prototype.IsMVC;
    /** @type {?} */
    TXSettings.prototype.NoPrintEndpoint;
    /** @type {?} */
    TXSettings.prototype.NoFileSystemAccess;
    /** @type {?} */
    TXSettings.prototype.ConnectionID;
    /** @type {?} */
    TXSettings.prototype.DocumentTargetMarkers;
    /** @type {?} */
    TXSettings.prototype.BackColor;
    /** @type {?} */
    TXSettings.prototype.StatusBarColor;
    /** @type {?} */
    TXSettings.prototype.EditMode;
    /** @type {?} */
    TXSettings.prototype.ContextMenusEnabled;
    /** @type {?} */
    TXSettings.prototype.SaveDocumentQuestionDialogEnabled;
    /** @type {?} */
    TXSettings.prototype.TableGridLines;
    /** @type {?} */
    TXSettings.prototype.TextFrameMarkerLines;
    /** @type {?} */
    TXSettings.prototype.ControlChars;
    /** @type {?} */
    TXSettings.prototype.TextFieldsEditable;
    /** @type {?} */
    TXSettings.prototype.FormattingPrinter;
    /** @type {?} */
    TXSettings.prototype.Culture;
    /** @type {?} */
    TXSettings.prototype.UICulture;
    /** @type {?} */
    TXSettings.prototype.UserNames;
    /** @type {?} */
    TXSettings.prototype.ReconnectTimeout;
    /** @type {?} */
    TXSettings.prototype.WebSocketURL;
}
